src/Columns/ColumnVector.cpp

```diff
@@ -135,6 +137,10 @@ void ColumnVector<T>::getPermutation(bool reverse, size_t limit, int nan_directi
     }
     else
     {
+        /// If bitonic sort if specified as preferred than `nan_direction_hint` equals specific value 42.
+        if (nan_direction_hint == 42 && BitonicSort::getInstance().sort(data, res, !reverse))
```

Comment:
It's OK for now, but it's better to make enum and describe it in nan_direction_hint comment in IColumn.h. I'll make it later.